### ISSUE TITLE
ClickHouse: == equality operator is unparsable

### ISSUE THREAD

**Issue by friedrich123456789** [sqlfluff/sqlfluff#8033]
### What happened

ClickHouse accepts `==` as a synonym for `=` (both map to `equals()`).
The `clickhouse` dialect reports `==` as an unparsable section.

### Expected behaviour

`a == b` parses as an equality comparison (equivalent to `a = b`).

### Observed behaviour

```
L1:P25 PRS: Found unparsable section: '== 1'
```

### How to reproduce

```sql
SELECT a FROM t WHERE a == 1
```

```bash
sqlfluff parse --dialect clickhouse -
```

Valid, verified on ClickHouse 26.4.2.
Docs: https://clickhouse.com/docs/sql-reference/operators (comparison operators).

### Dialect

clickhouse

### Version

sqlfluff 4.2.2 (latest release), Python 3.12; also applies to `main`.

### Configuration

None — default rules, `--dialect clickhouse`.
